src/bootstrap/lib.rs

```diff
@@ -482,6 +482,12 @@ impl Build {
                   .env("RUSTC_DEBUGINFO_LINES", self.config.rust_debuginfo_lines.to_string());
         }
 
+        // Since the metadata of dependencies can conflict, we're adding a
+        // special suffix to make it unique.
+        if mode == Mode::Librustc {
```

Comment:
Could this actually be for `mode != Mode::Tool` (e.g. also apply to libstd/libtest)